src/Common/Commands.Common/AzurePSCmdlet.cs

```diff
@@ -682,22 +682,78 @@ protected override void ProcessRecord()
         /// <summary>
         /// Job Name paroperty iof this cmdlet is run as a job
         /// </summary>
         public virtual string ImplementationBackgroundJobDescription
         {
             get
             {
-                string name = "Long Running Azure Operation";
-                string commandName = MyInvocation?.MyCommand?.Name;
-                if (!string.IsNullOrWhiteSpace(commandName))
+                if (_implementationBackgroundJobDescription != null)
                 {
-                    name = string.Format("Long Running Operation for '{0}'", commandName);
+                    return _implementationBackgroundJobDescription;
                 }
+                else
+                {
+                    string name = "Long Running Azure Operation";
+                    string commandName = MyInvocation?.MyCommand?.Name;
+                    string objectName = null;
+                    if (MyInvocation?.BoundParameters?.ContainsKey("Name") == true)
+                    {
+                        objectName = MyInvocation.BoundParameters["Name"].ToString();
+                    }
+                    else if (MyInvocation?.BoundParameters?.ContainsKey("InputObject") == true)
+                    {
+                        var type = MyInvocation.BoundParameters["InputObject"].GetType();
+                        var inputObject = Convert.ChangeType(MyInvocation.BoundParameters["InputObject"], type);
+                        if (type.GetProperty("Name") != null)
+                        {
+                            objectName = inputObject.GetType().GetProperty("Name").GetValue(inputObject).ToString();
+                        }
+                        else if (type.GetProperty("ResourceId") != null)
+                        {
+                            string[] tokens = inputObject.GetType().GetProperty("ResourceId").GetValue(inputObject).ToString().Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                            if (tokens.Length >= 8)
+                            {
+                                objectName = tokens[tokens.Length - 1];
+                            }
+                        }
+                    }
+                    else if (MyInvocation?.BoundParameters?.ContainsKey("ResourceId") == true)
+                    {
+                        string[] tokens = MyInvocation.BoundParameters["ResourceId"].ToString().Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                        if (tokens.Length >= 8)
+                        {
+                            objectName = tokens[tokens.Length - 1];
+                        }
+                    }
+
+                    if (!string.IsNullOrWhiteSpace(commandName))
+                    {
+                        if (!string.IsNullOrWhiteSpace(objectName))
+                        {
+                            name = string.Format("Long Running Operation for '{0}' : '{1}'", commandName, objectName);
```

Comment:
@maddieclayton we might want to tweak this to be `Long Running Operation for '{0}' on resource '{1}'`